tests/test_app/test_composable.py

```diff
@@ -634,5 +633,34 @@ def main(self, val: int) -> int:
     __app_registry.pop(get_object_provenance(app_decorated_first1))
 
 
+# have to define this at module level for pickling to work
+@define_app
+def func2app(arg1: int, exponent: int) -> float:
+    return arg1 ** exponent
+
+
+def test_decorate_app_function():
+    """works on functions now"""
+    import inspect
+
+    sqd = func2app(exponent=2)
+    assert sqd(3) == 9
```

Comment:
What is role of the `main` function here?

Reply:
`main()` method (invoked within `__call__()`) is assembling the arguments for the call to the bound static method (the original function being decorated)